### ISSUE TITLE
data/request api failing on SiteWhitelist

### ISSUE THREAD

**Issue by vlimant** [dmwm/WMCore#8161]
https://cmsweb.cern.ch/reqmgr2/data/request?mask=SiteWhitelist&status=assigned&status=acquired&status=running-open&status=running-closed 

succeeds, while 

https://cmsweb.cern.ch/reqmgr2/data/request?mask=SiteWhitelist&status=assigned&status=acquired&status=running-open&status=running-closed&status=completed

goes into 

500 Internal Server Error

Server error

**Comment by amaltaro** [comment]
This workflow:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=wmagent_dmason_HIG-RunIISummer15wmLHEGS-00744_161213_065940_2749

is one example that makes ReqMgr2 to raise an exception, with this traceback [1].

So, this workflow has SiteWhitelist both at top level and at Step level (old'ish request spec, nowadays it's only supported at top level). The masking code creates and empty list and starts appending the Step/Task values to it (which are another list object):
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/ReqMgr/Service/Request.py#L230

but in the very end, it calls the set builtin to get unique values, which then fails since a list object cannot be hashed (L232).

In order to get that API working, we can either announce or reject that workflow. Then a proper fix can be made for the next release.

[1]
```
[13/Sep/2017:12:27:22]  SERVER OTHER ERROR __builtins__.TypeError 51fdff98e4c6bedb9e4602347fad38bc (unhashable type: 'list')
[13/Sep/2017:12:27:22]    Traceback (most recent call last):
[13/Sep/2017:12:27:22]      File "/data/srv/beHG1709d/sw/slc6_amd64_gcc493/cms/reqmgr2/1.1.6.pre11/lib/python2.7/site-packages/WMCore/REST/Server.py", line 711, in default
[13/Sep/2017:12:27:22]        return self._call(RESTArgs(list(args), kwargs))
[13/Sep/2017:12:27:22]      File "/data/srv/beHG1709d/sw/slc6_amd64_gcc493/cms/reqmgr2/1.1.6.pre11/lib/python2.7/site-packages/WMCore/REST/Server.py", line 787, in _call
[13/Sep/2017:12:27:22]        obj = apiobj['call'](*safe.args, **safe.kwargs)
[13/Sep/2017:12:27:22]      File "/data/srv/beHG1709d/sw/slc6_amd64_gcc493/cms/reqmgr2/1.1.6.pre11/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 342, in get
[13/Sep/2017:12:27:22]        result = self._mask_result(mask, result)
[13/Sep/2017:12:27:22]      File "/data/srv/beHG1709d/sw/slc6_amd64_gcc493/cms/reqmgr2/1.1.6.pre11/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 249, in _mask_result
[13/Sep/2017:12:27:22]        self._maskTaskStepChain(masked_result[req_name], req_info, "Step", mask_key)
[13/Sep/2017:12:27:22]      File "/data/srv/beHG1709d/sw/slc6_amd64_gcc493/cms/reqmgr2/1.1.6.pre11/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 232, in _maskTaskStepChain
[13/Sep/2017:12:27:22]        masked_dict[mask_key] = list(set(masked_dict[mask_key]))
[13/Sep/2017:12:27:22]    TypeError: unhashable type: 'list'
[13/Sep/2017:12:27:22] vocms0136.cern.ch 2001:1458:202:44::102:630c "GET /reqmgr2/data/request?mask=SiteWhitelist&status=closed-out HTTP/1.1" 500 Internal Server Error [data: 4687 in 743 out 266654 us ] [auth: OK "/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=vlimant/CN=652956/CN=Jean-Roch Vlimant" "" ] [ref: "" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:55.0) Gecko/20100101 Firefox/55.0" ]
```

**Comment by amaltaro** [comment]
Works for me now. It seems there was an incompatibility problem between old request spec and new (new as of 2 or 3 years ago). I'm closing this issue too.
